src/libcore/num/mod.rs

```diff
@@ -1765,7 +1765,11 @@ assert_eq!((-a).mod_euc(-b), 1);
             pub fn mod_euc(self, rhs: Self) -> Self {
                 let r = self % rhs;
                 if r < 0 {
-                    r + rhs.abs()
+                    if rhs.is_negative() {
```

Comment:
This line and the previous might as well be consistent (whichever of `< 0` or `is_negative` you prefer).